website/docs/getting-started/cluster-setup/kubernetes.md

```diff
@@ -1,8 +1,117 @@
 ---
 title: Kubernetes
-sidebar_position: 2
+sidebar_position: 1
 ---
 
 # Setup a minikube cluster
+Please note that this documentation is only useful in setting up a development environment, it is not recommended for a production environment.
 
-Please make sure you have read [this document](development-clusters.md) first.
\ No newline at end of file
+This guide assumes that you have [installed minikube](https://minikube.sigs.k8s.io/docs/start/) on your system.
+
+If you are using a Kubernetes cluster other than minikube, this guide assumes that you have admin privileges to the cluster and are logged in with the admin user; operator installation is only possible with an admin user.
+
+**Agenda for this guide:**
+* [Install the OLM](#install-the-olm)
+* [Install the Service Binding Operator](#install-the-service-binding-operator)
+* [Install an operator](#install-an-operator)
+* [Verify the operator installation](#verify-the-operator-installation)
+* [Enable Ingress addon](#enable-ingress-addon)
+
+## Install the OLM
+The Operator Lifecycle Manager(OLM) is a component of the Operator Framework, an open source toolkit to manage Kubernetes native applications, called Operators, in a streamlined and scalable way.[(Source)](https://olm.operatorframework.io/)
```

Comment:
Shouldn't we put this in the Operator doc?

Reply:
Yes we should. But at the moment, we are adding this doc. So we should put it here. If operator doc was ready, we could have linked to it. Once we have it ready, we could replace this paragraph with a link to it.

Reply:
Rather than coming up with our own definition of what Operator is we should link to existing Kubernetes documentation 

https://kubernetes.io/docs/concepts/extend-kubernetes/operator/

Reply:
We can and should surely link to k8s docs whenever it makes sense. But we should also make things simpler to understand for users who don't want/need to know things from k8s perspective.

Having said that, in this case, the `Motivatoin` section of the doc is pretty good and can be used to explain Operators instead of using what I suggested.

But overall, I hope that odo docs attempt to make k8s things simpler to understand for users. If that means coming up with our own definitions, so be it. Code should not be the only place where odo follows the philosophy of making k8s abstractions simple for users.

Reply:
>  Code should not be the only place where odo follows the philosophy of making k8s abstractions simple for users.

agreed 

> If that means coming up with our own definitions,

That is dangerous. We should not come up with our own definitions for stuff that we did no created. We should no try to redefine things.

Reply:
Why not use already existing definition from  https://kubernetes.io/docs/concepts/extend-kubernetes/operator/ or https://www.redhat.com/en/topics/containers/what-is-a-kubernetes-operator ? Do you think that those are too technical?

Reply:
> That is dangerous. 

The other way of looking at it is trying to do ELI5. Instead of really trying to do ELI5, we should try to explain things as if our readers have no knowledge of k8s.

> Why not use already existing definition from https://kubernetes.io/docs/concepts/extend-kubernetes/operator/ or https://www.redhat.com/en/topics/containers/what-is-a-kubernetes-operator ? Do you think that those are too technical?

I already said in my previous comment (https://github.com/openshift/odo/pull/4973#discussion_r686613855) that in this case, reusing what's in the "Motivation" section makes sense because it's written in really simple language.

Reply:
I will add the excerpt of Motivation from the official k8s doc, it's short and simple as compared to the RH article.